op-service/queue/queue.go

```diff
@@ -0,0 +1,25 @@
+package queue
```

Comment:
Curious what the benefit of this over using the standard library like https://pkg.go.dev/container/list@go1.23.1

Reply:
I think what I am trying to drive towards here is using an interface to interact with the queue rather than committing to a specific implementation. 

That library implements a doubly linked list. That might be more optimal than the slice implementation I started with here, since we do not need random access to the queue of blocks and we are mostly adding to the front and the back of the queue. 

OTOH it could be a premature optimization, since it will probably make writing tests a little more difficult and I don't think performance is particularly critical here (versus simplicitly / readability etc).

Reply:
I also think the std `list` has a couple of disadvantages over a more direct queue implementation
* it's a linked list, but we don't need arbitrary insertion and deletion, but rather only insert at both ends of the queue, and pop only from the front of the queue. So a barebones slice better fits the situation of the queue.
* the std `list` works with generic `Element`s, which in turn have a `Value any` field, so it would be much more cumbersome and type-unsafe to work with it
* `list` also doesn't have methods for `pop`ping, you'd always need to access the list front and delete in two steps, which is more natural to happen in a single step with a queue. Admittedly, the way we use the queue in the batcher, we actually do first only access the front of the queue, and only after successful addition to the channel builder also remove the front element.